### PR TITLE
adding annotation with hash value of config map to restart pods

### PR DESCRIPTION
**Pull request by ishaan-tf**

### CODE REVIEW

File: charts/tfy-notebook-controller/README.md [hunk 1]
```diff
@@ -4,7 +4,61 @@ This Helm chart package, provided by TrueFoundry, contains configurations and re
 
 ## Parameters for tfy-notebook-controller
 
-############## Parameters for tfy-notebook-controller
+############### Parameters for tfy-notebook-controller
```
Comment (debajyoti-truefoundry) [inline charts/tfy-notebook-controller/README.md:7]: There should not be any changes in this file.

File: charts/tfy-notebook-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -18,6 +18,8 @@ spec:
         {{- with .Values.labels }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```
Comment (debajyoti-truefoundry) [inline charts/tfy-notebook-controller/templates/deployment.yaml:22]: You have to truncate it to 63 characters.

Comment (debajyoti-truefoundry) [reply charts/tfy-notebook-controller/templates/deployment.yaml:22]: Do we need to add quote here?

File: charts/tfy-notebook-controller/README.md (hunk 1 above)
Comment (ishaan-tf) [reply charts/tfy-notebook-controller/README.md:7]: this was autogenerated by github actions

File: charts/tfy-notebook-controller/templates/deployment.yaml (hunk 2 above)
Comment (debajyoti-truefoundry) [reply charts/tfy-notebook-controller/templates/deployment.yaml:22]: > include (print $.Template.BasePath "/configmap.yaml")

Will this just take the template or will this take the rendered configmap yaml?

Comment (debajyoti-truefoundry) [inline charts/tfy-notebook-controller/templates/deployment.yaml:22]:
```suggestion
        truefoundry.com/configChecksum: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```

Comment (ishaan-tf) [reply charts/tfy-notebook-controller/templates/deployment.yaml]: it takes the rendered configmap yaml here

File: charts/tfy-notebook-controller/templates/deployment.yaml [hunk 3]
```diff
@@ -19,7 +19,7 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
       annotations:
-        truefoundry.com/configChecksum: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum | trunc 63 }}
+        truefoundry.com/configChecksum: "{{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum | trunc 63 }}"
```
Comment (debajyoti-truefoundry) [inline charts/tfy-notebook-controller/templates/deployment.yaml:22]: Use the quote function

File: charts/tfy-notebook-controller/templates/deployment.yaml [hunk 4]
```diff
@@ -18,6 +18,8 @@ spec:
         {{- with .Values.labels }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
+      annotations:
```
Comment (debajyoti-truefoundry) [inline charts/tfy-notebook-controller/templates/deployment.yaml:21]: Please bump the chart version.

**Review by debajyoti-truefoundry — APPROVED** [review]

**Pull request merged**